File: app/src/main/java/com/example/typego/AchievementsFragment.java
package com.example.typego;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.typego.adapters.AchievementsAdapter;
import com.example.typego.utils.Achievement;
import com.example.typego.utils.User;

import java.util.ArrayList;

public class AchievementsFragment extends Fragment {


    public AchievementsFragment() {

    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_achievements, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        User currentUser = User.getFromStoredData(getContext());
        ArrayList<Achievement> achievements = currentUser.getAchievements();
        RecyclerView rvAchievements = view.findViewById(R.id.rv_achievements);
        rvAchievements.setAdapter(new AchievementsAdapter(
                getContext(),
                achievements,
                User.getFromStoredData(getContext())));
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        rvAchievements.setLayoutManager(linearLayoutManager);
    }
}

File: app/src/main/java/com/tetsoft/typego/achievements/presentation/AchievementsFragment.kt
package com.tetsoft.typego.achievements.presentation

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.hilt.navigation.fragment.hiltNavGraphViewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.tetsoft.typego.R
import com.tetsoft.typego.databinding.FragmentAchievementsBinding
import com.tetsoft.typego.core.ui.BaseFragment

class AchievementsFragment : BaseFragment<FragmentAchievementsBinding>() {

    private val viewModel : AchievementsViewModel by hiltNavGraphViewModels(R.id.main_navigation)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.rvAchievements.adapter = AchievementsAdapter(
            requireContext(),
            viewModel.getAchievementsList(),
            viewModel.getHistory(),
            viewModel.getAchievementsProgressList()
        )
        val linearLayoutManager = LinearLayoutManager(context)
        binding.rvAchievements.layoutManager = linearLayoutManager
    }

    override fun initBinding(
        inflater: LayoutInflater,
        container: ViewGroup?
    ): FragmentAchievementsBinding {
        return FragmentAchievementsBinding.inflate(inflater, container, false)
    }
}